src/components/DeckMechanisms/handleCardEffect.js

```diff
@@ -298,8 +301,24 @@ const handleCardEffect = (state, card, mode) => {
 
     // Harvesters of Souls
     case 'N38': {
-      const copiedCard = getHarvestersOfSoulsCopiedCard(state, card.level)
-      state.deck.push(copiedCard)
+      // The RNG for Harvesters of Souls is determined by first choosing a level for the
+      // created copy and then creating the copy if the level is greater than 1 (otherwise
+      // Harvesters weren't able to copy the card)
+      const lowestPossibleLevel =
+        card.level + HARVESTERS_OF_SOULS_RNG.LEVEL_BONUS[state.RNG] - 2
+      const possibleLevelValues = Array.from(
+        { length: 2 * HARVESTERS_OF_SOULS_RNG.MAX_DEVIATION + 1 },
+        (_, i) => lowestPossibleLevel + i
+      )
+      const level = arrayRandom(possibleLevelValues)
+      if (level >= 1) {
+        const copiedCard = getHarvestersOfSoulsCopiedCard(
```

Comment:
I think all this logic should be in `getHarvestersOfSoulsCopiedCard`. And if the function return `null` (or `undefined`, it doesn’t matter much), we don’t push anything into the deck.

Reply:
Thanks ;)